src/khoj/routers/api.py

```diff
@@ -332,6 +332,31 @@ async def update_chat_model(
     return {"status": "ok"}
 
 
+@api.post("/config/data/search/model", status_code=200)
+@requires(["authenticated"])
+async def update_search_model(
+    request: Request,
+    id: str,
+    client: Optional[str] = None,
+):
+    user = request.user.object
+
+    new_config = await adapters.aset_user_search_model(user, int(id))
+
+    update_telemetry_state(
+        request=request,
+        telemetry_type="api",
+        api="set_search_model",
+        client=client,
+        metadata={"search_model": new_config.setting.name},
+    )
+
+    if new_config is None:
+        return {"status": "error", "message": "Model not found"}
```

Comment:
Should the `update_telemetry_state` be done only after/if new_config is not None